code/modules/heavy_vehicle/mech_interaction.dm

```diff
@@ -1,7 +1,7 @@
 /mob/living/MouseDrop(atom/over)
 	if(usr == src && usr != over)
 		if(istype(over, /mob/living/heavy_vehicle))
-			if(usr.mob_size >= MOB_SMALL && usr.mob_size <= 14)
+			if((usr.mob_size >= MOB_SMALL && usr.mob_size <= 14) || HAS_TRAIT(usr, TRAIT_MECH_PILOT))
```

Comment:
No, if you use the trait, use only the trait; assign to relevant mobs the mech_pilot trait and remove the mob size check